v22.2/fips-compliance.md

```diff
@@ -0,0 +1,162 @@
+toc: true
+docs_area: deploy
+---
+
+## Overview of FIPS 140-2 compliance in CockroachDB
+
+{{site.data.alerts.callout_info}}
+FIPS-compliant builds of CockroachDB are **experimental** and are not yet qualified for production use.
+{{site.data.alerts.end}}
+
+[Federal Information Processing Standards (FIPS) 140-2](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.140-2.pdf) is a mandatory standard for the protection of sensitive or valuable data within systems used by and for United States and Canadian federal government agencies. In the United States, the standard is applicable to all federal agencies that use cryptographic-based security systems to protect sensitive information in computer and telecommunication systems (including voice systems) as defined in Section 5131 of the Information Technology Management Reform Act of 1996, Public Law 104-106 and the Federal Information Security Management Act of 2002, Public Law 107-347.
+
+When a cryptographic module or library has a FIPS 140-2 certificate, it has been tested and formally validated by the U.S. and Canadian Governments.
+
+The standard CockroachDB binaries rely on the Go language for cryptographic functions and do not use OpenSSL libraries installed in the operating system. Go's cryptographic libraries are not FIPS-validated.
+
+FIPS-compliant CockroachDB binaries are built using [Red Hat's Go FIPS with OpenSSL toolchain](https://github.com/golang-fips/go), which contains the necessary modifications for the Go crypto library to use an external cryptographic library in a FIPS compliant way. FIPS-compliant CockroachDB runtimes use the host operating system's OpenSSL libraries and commands, rather than Go's cryptographic libraries. When the installed OpenSSL has a FIPS 140-2 certificate and [FIPS mode is enabled in the Linux kernel](#enable-fips-mode-in-the-linux-kernel), the CockroachDB runtime is compliant with FIPS 140-2. FIPS-compliant binaries and Docker images are available for CockroachDB v22.2.7 and above. CockroachDB FIPS-compliant runtimes run only on Intel 64-bit Linux systems.
+
+### FIPS-compliant features
+
+When you use a FIPS-compliant CockroachDB runtime and each cluster node's OpenSSL is FIPS-validated and configured correctly, the following features have been tested to ensure that their cryptographic operations comply with FIPS 140-2:
+
+- [Encryption At Rest](security-reference/encryption.html#encryption-keys-used-by-cockroachdb-self-hosted-clusters)
+- [Encrypted Backups](take-and-restore-encrypted-backups.html)
+- [Change Data Capture to Kafka over TLS](create-changefeed.html#query-parameters)
+- [Certificate-based Node-to-Node and Client-to-Node Authentication](authentication.html)
+- [SASL SCRAM-SHA-256 Password Authentication](security-reference/scram-authentication.html)
+- [Built-in SQL cryptographic functions](functions-and-operators.html#cryptographic-functions)
```

Comment:
Change this list to a table, and be sure to include rows that list all the other CockroachDB features that 1) use cryptography 2) do not delegate said crypto to OpenSSL, and thus cannot be part of a FIPS story.
Something like the following:

| CockroachDB feature | Cryptography pass-through to OpenSSL | Usage in SQL apps compatible with FIPS compliance for the entire deployment |
|--|--|--|
| Encryption at rest | Yes | Yes |
| Encrypted backups | Yes | Yes |
| CDC to Kafka over TLS | Yes | Yes |
| Async binary replication over TLS (v23.1 only, TECHNOLOGY PREVIEW) | Yes | Yes |
| Node-to-node TLS | Yes | Yes |
| Client-to-node TLS | Yes | Yes |
| SCRAM password authn for SQL | Yes | Yes |
| Bcrypt-based password authn for SQL | No | No |
| OIDC authn for DB Console and HTTP APIs | Yes (NOTE TO DOCS: PLEASE DOUBLE CHECK) | Yes (PLEASE DOUBLE CHECK) |
| GSSAPI/Kerberos authn for SQL | No | No |
| SQL `pgcrypto` compatibility functions | Yes (see notes below) | Depends on application usage (see notes below) |
| SQL hashing functions like `sha256`, `md5` etc | See notes below | See notes below |

(btw we also need a section that talks about hashing functions. I don't see i currently.)

Reply:
@abhinavg6 WDYT about this suggestion?

Reply:
We can revisit this later if you still want this change, but I am out of time for now. Thanks for the feedback.